throttle.js

```diff
@@ -41,13 +51,6 @@ export default function (delay, noTrailing, callback, debounceMode) {
 		cancelled = true;
 	}
 
-	// `noTrailing` defaults to falsy.
-	if (typeof noTrailing !== 'boolean') {
```

Comment:
@seiyab why do we need to remove this part?

Reply:
@niksy 
I think the purpose of this part is handling the optional argument `noTrailing` at the intermediate position.
When we have an optional argument at intermediate position, we must shift arguments that follow the optional one.
```js
// Previous declaration
function throttle(delay, noTrailing, callback, debounceMode) {
  // ...
}

const delay = 100;
const noTrailing = true;
const callback = function() { return };
const debounceMode = undefined;

// Call with full arguments is straightforward.
throttle(delay, noTrailing, callback, debounceMode);

// Skipping an optional argument is not straightforward.
// The actual argument `callback` is passed at the position of the formal argument `noTrailing`.
// The actual argument `debouceMode` is passed at the position of the formal argument `callback`.
// So throttle should shift arguments, on the other words, it should handle noTrailing as callback, callback as debounceMode.
throttle(delay, callback, debounceMode);

// Actual: throttle(delay, callback,   debounceMode)
//                    |      |            |
//                    v      v            v
// Formal: throttle(delay, noTrailing, callback,   debounceMode) 
```
The terms "actual argument" and "formal argument" refer to ones as described [here](https://overiq.com/c-programming-101/actual-and-formal-arguments-in-c/).

In this PR, I rearrange the arguments as discussed in https://github.com/niksy/throttle-debounce/issues/34 .
So the actual argument `callback` is always passed at the position of the formal argument `callback`.
Then we don't need the part.
It doesn't change the behavior excluding how to pass options, as far as I know.
(The way to pass options changes because I remove the duplicate).

Reply:
No need, this is understandable.